File: src/ads_runner_client/lib/domain/entities/media_asset.dart
import 'package:equatable/equatable.dart';

class MediaAsset extends Equatable {
  final String id;
  final String fileName;
  final String fileUrl;
  final String fileType;
  final int fileSize;
  final DateTime uploadedAt;
  final String? thumbnailUrl;

  const MediaAsset({
    required this.id,
    required this.fileName,
    required this.fileUrl,
    required this.fileType,
    required this.fileSize,
    required this.uploadedAt,
    this.thumbnailUrl,
  });

  @override
  List<Object?> get props => [id, fileName, fileUrl];
}

File: src/AdsRunner.Domain/Entities/MediaAsset.cs
using AdsRunner.Domain.Common;

namespace AdsRunner.Domain.Entities;

public class MediaAsset : BaseEntity
{
    public string FileName { get; set; } = string.Empty;
    public string FileUrl { get; set; } = string.Empty;
    public string FileType { get; set; } = string.Empty;
    public long FileSize { get; set; }
    public DateTime UploadedAt { get; set; }
    public string? ThumbnailUrl { get; set; }

    // Foreign keys
    public Guid CampaignId { get; set; }

    // Navigation properties
    public Campaign Campaign { get; set; } = null!;
}
